src/utils/inject-stylesheet.js

```diff
@@ -7,13 +11,56 @@
  *
  * @returns {void}
  */
-export function injectStylesheet(endpoint, env, version, uid, cid) {
-	const queryParams = (cid) ? '?previewcid=' + cid : '';
+export function injectStylesheetWithAutoReplace(endpoint, env, version, uid, cid, attempt = 0) {
+	injectStylesheet(endpoint, env, version, uid, cid)
+	const url = getStyleSheetUrl(endpoint, env, version, uid, cid);
+	const cssFile = getCSSFile(url);
+
+	if (cssFile && window.MutationObserver) {
+		const observer = new window.MutationObserver(function (mutationList) {
+			mutationList.forEach(function (mutation) {
+				mutation.removedNodes.forEach(function (node) {
+					if (node.nodeName === 'LINK' && node.href === url && attempt < MAX_RETRIES) {
+						console.warn('Evolv AI: CSS removed, re-adding');
+						observer.disconnect();
+						injectStylesheetWithAutoReplace(endpoint, env, version, uid, cid, attempt + 1);
+					}
+				});
+			});
+		});
+		observer.observe(getCSSFile(url).parentNode, {
+			childList: true
+		});
+
+		setTimeout(function () {
+			observer.disconnect();
```

Comment:
It's okay to call this on an already disconnected observer right?

Reply:
just tested and it doesn't error